src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -73,6 +75,26 @@ internal static MsalServiceException FromHttpResponse(
             return ex;
         }
 
+        private static string GetErrorMessage(string errorMessage, HttpResponse httpResponse, RequestContext context)
+        {
+            if (httpResponse.StatusCode == HttpStatusCode.NotFound && context != null)
+            {
+                errorMessage += "\nAuthority used: " + context.ServiceBundle.Config.Authority?.AuthorityInfo?.CanonicalAuthority?.AbsoluteUri?.Split('?')[0];
```

Comment:
```suggestion
                errorMessage += "\nAuthority used: " + context.ServiceBundle.Config.Authority?.AuthorityInfo?.CanonicalAuthority?.AbsoluteUri?.Split('?')[0];
```

you can use `StringBuilder` for efficiency:

```csharp
private static string GetErrorMessage(string errorMessage, HttpResponse httpResponse, RequestContext context)
{
    // Using StringBuilder for more efficient string concatenation
    var sb = new StringBuilder(errorMessage);

    if (httpResponse.StatusCode == HttpStatusCode.NotFound && context != null)
    {
        sb.Append("\nAuthority used: ")
          .Append(context.ServiceBundle.Config.Authority?.AuthorityInfo?.CanonicalAuthority?.AbsoluteUri?.Split('?')[0]);
        
        if (!context.ApiEvent.TokenEndpoint.IsNullOrEmpty())
        {
            sb.Append("\nToken Endpoint: ")
              .Append(context.ApiEvent?.TokenEndpoint);
        }

        if (!context.ServiceBundle.Config.AzureRegion.IsNullOrEmpty())
        {
            sb.Append("\nRegion Used: ")
              .Append(context.ServiceBundle.Config.AzureRegion);
        }
    }

    return sb.ToString();
}
```

Reply:
you can also use `AppendLine`